AdobeStockImageApi/Api/SaveImageInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImageApi\Api;
+
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NotFoundException;
+
+/**
+ * SaveImageInterface
+ *
+ * @api
+ */
+interface SaveImageInterface
+{
+    /**
+     * Runs the save image process.
+     *
+     * @param int    $adobeId
+     * @param string $destinationPath
+     *
+     * @return void
+     * @throws CouldNotSaveException
+     * @throws NotFoundException
+     * @throws LocalizedException
+     */
+    public function execute(int $adobeId, string $destinationPath): void;
```

Comment:
I think the service accepting URL instead of adobeId could be more universal.

Reply:
That's a great catch. However, we still need adobeID for getting/saving the asset. I decided to pass the `AssetInterface` with URL instead.
In that way, we can eliminate two interfaces and implementations for saving image/preview and use the unified one. 
Please, take a look at the recent commit.